shell/middleware/authenticated.js

```diff
@@ -1,210 +1,55 @@
-    }
-
-    const res = store.getters['management/byId'](MANAGEMENT.SETTING, SETTING.FIRST_LOGIN);
-    const plSetting = store.getters['management/byId'](MANAGEMENT.SETTING, SETTING.PL);
-
-    firstLogin = res?.value === 'true';
+  const initialSetupSettings = await store.dispatch('fetchRancherInitialSetupSettings');
 
-    if (!initialPass && plSetting?.value === 'Harvester') {
-      initialPass = 'admin';
-    }
-  } catch (e) {
+  // Set the favicon - use custom one from store if set
+  if (!haveSetFavIcon()) {
+    setFavIcon(store);
   }
 
-  if ( firstLogin === null ) {
-    try {
-      const res = await store.dispatch('rancher/find', {
-        type: 'setting',
-        id:   SETTING.FIRST_LOGIN,
-        opt:  { url: `/v3/settings/${ SETTING.FIRST_LOGIN }` }
-      });
-
-      firstLogin = res?.value === 'true';
-
-      const plSetting = await store.dispatch('rancher/find', {
-        type: 'setting',
-        id:   SETTING.PL,
-        opt:  { url: `/v3/settings/${ SETTING.PL }` }
-      });
-
-      if (!initialPass && plSetting?.value === 'Harvester') {
-        initialPass = 'admin';
-      }
-    } catch (e) {
-    }
-  }
-
-  // TODO show error if firstLogin and default pass doesn't work
-  if ( firstLogin ) {
-    const ok = await tryInitialSetup(store, initialPass);
-
-    if (ok) {
-      if (initialPass) {
-        store.dispatch('auth/setInitialPass', initialPass);
-      }
-
-      return redirect({ name: 'auth-setup' });
-    } else {
-      return redirect({ name: 'auth-login' });
-    }
-  }
-
-  // Make sure you're actually logged in
-  function isLoggedIn(me) {
-    store.commit('auth/hasAuth', true);
-    store.commit('auth/loggedInAs', me.id);
-  }
-
-  function notLoggedIn() {
-    store.commit('auth/hasAuth', true);
-
-    if ( route.name === 'index' ) {
-      return redirect(302, '/auth/login');
-    } else {
-      return redirect(302, `/auth/login?${ TIMED_OUT }`);
-    }
-  }
-
-  function noAuth() {
-    store.commit('auth/hasAuth', false);
+  // TODO show error if initialSetupSettings.isFirstLogin and default pass doesn't work
+  if ( initialSetupSettings.isFirstLogin ) {
+    await store.dispatch('auth/handleFirstLogin', initialSetupSettings.defaultPassword);
   }
 
   if ( store.getters['auth/enabled'] !== false && !store.getters['auth/loggedIn'] ) {
-    // `await` so we have one successfully request whilst possibly logged in (ensures fromHeader is populated from `x-api-cattle-auth`)
-    await store.dispatch('auth/getUser');
-
-    const v3User = store.getters['auth/v3User'] || {};
-
-    if (v3User?.mustChangePassword) {
-      return redirect({ name: 'auth-setup' });
-    }
+    await store.dispatch('auth/handleUserMustChangePassword');
 
     // In newer versions the API calls return the auth state instead of having to make a new call all the time.
     const fromHeader = store.getters['auth/fromHeader'];
 
     if ( fromHeader === 'none' ) {
-      noAuth();
+      store.dispatch('auth/noAuth');
```

Comment:
Could we turn this whole chain of conditions into a switch for clarity?

Reply:
Moved to a switch